### ISSUE TITLE
Mockk cannot handle sealed classes in `every` block

### ISSUE THREAD

**Issue by ThanksForAllTheFish** [mockk/mockk#934]
Sample project: https://github.com/ThanksForAllTheFish/mockk832
Related: https://github.com/mockk/mockk/issues/832

Mockk 1.13.2 still throws

```
org.t4atf.mockk832.Node
java.lang.InstantiationError: org.t4atf.mockk832.Node
	at jdk.internal.reflect.GeneratedSerializationConstructorAccessor17.newInstance(Unknown Source)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480)
	at org.objenesis.instantiator.sun.SunReflectionFactoryInstantiator.newInstance(SunReflectionFactoryInstantiator.java:48)
	at io.mockk.proxy.jvm.ObjenesisInstantiator.instanceViaObjenesis(ObjenesisInstantiator.kt:75)
	at io.mockk.proxy.jvm.ObjenesisInstantiator.instance(ObjenesisInstantiator.kt:42)
	at io.mockk.impl.instantiation.JvmInstantiator$instantiate$2.invoke(JvmInstantiator.kt:16)
	at io.mockk.impl.instantiation.AbstractInstantiator.instantiateViaInstanceFactoryRegistry(AbstractInstantiator.kt:17)
```

when using sealed classes on kotlin 1.7 and java 17. The issue now is in `every` blocks, for instance:

```
class Root : Node()
class Leaf : Node()

interface Factory {
    fun create() : Node
    fun copy(node: Node) : Node
}

open class MyTest {
    @Test
    open fun aTest() {
        val mockk = mockk<Factory>()
        every { mockk.create() } returns Leaf() // this was fixed in mockk832

        println(mockk.create())

        every { mockk.copy(any()) } returns Root() //but this still breaks
    }
}
```

**Comment by aSemy** [comment]
Looks like a good demo - can you (or someone else) update https://github.com/mockk/mockk/blob/6d5fe10cfd3662c7716f7381ecc6b06bde12a091/modules/mockk/src/commonTest/kotlin/io/mockk/it/SealedClassTest.kt and make a PR?

**Comment by diastremskii** [comment]
The test can be something like that:
<details>
<summary>Example test</summary>

```kotlin
class SealedInterfaceTest {

    @Test
    fun serviceTakesSealedInterfaceAsInput() {
        val mockNodeId = 1
        val factory = mockk<Factory> {
            every { getId(any()) } answers { mockNodeId }
        }

        val result = factory.getId(Root(0))

        assertEquals(mockNodeId, result)
    }

    companion object {

        sealed interface Node {
            val id: Int
        }

        data class Root(override val id: Int) : Node
        data class Leaf(override val id: Int) : Node

        interface Factory {
            fun create(): Node

            fun getId(node: Node): Int
        }

        class FactoryImpl : Factory {
            override fun create(): Node = Root(0)

            override fun getId(node: Node): Int = node.id
        }

    }
}
```

</details>
But it passes with current test setup since the issue is reproducible with JVM 17 bytecode. Should I still open a PR with a test if the test passes? 🤔 

One way to make it fail is to change JVM target [here](https://github.com/mockk/mockk/blob/c18440f9e34020c96caa49b994c7f3efe92c7c8a/buildSrc/src/main/kotlin/buildsrc/config/Deps.kt#L10) to `JavaVersion.VERSION_17` and run the tests with `-PjavaToolchainTestVersion=17`.


The exception itself seems to originate [here](https://github.com/mockk/mockk/blob/870bd59cbf3b0505172cd32dfa47d7eb3a797798/modules/mockk-agent/src/jvmMain/kotlin/io/mockk/proxy/jvm/ObjenesisInstantiator.kt#L57), so the problem is that we are trying to create a subclass with Byte Buddy, but we can't since we are creating a subclass for interface that is sealed.

**Comment by aSemy** [comment]
> Should I still open a PR with the test if the test passes? 🤔

yes please!

> One way to make it fail is to change JVM target [here](https://github.com/mockk/mockk/blob/c18440f9e34020c96caa49b994c7f3efe92c7c8a/buildSrc/src/main/kotlin/buildsrc/config/Deps.kt#L10) to `JavaVersion.VERSION_17` and run the tests with `-PjavaToolchainTestVersion=17`.

The toolchain for tests _should_ be overridden using the config here, but I'm not confident that it works as expected

https://github.com/mockk/mockk/blob/master/buildSrc/src/main/kotlin/buildsrc/convention/toolchain-jvm.gradle.kts

**Comment by odin-delrio** [comment]
Added the test for reproducing the error here https://github.com/mockk/mockk/pull/936 

It does fail in my laptop as well with JDK17

**Comment by cliffred** [comment]
Created a PR with a fix: #939
